tests/src/JIT/HardwareIntrinsics/X86/Shared/TestTable.cs

```diff
@@ -0,0 +1,56 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+//
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace HardwareIntrinsicTest
+{
+    public unsafe struct TestTable<T> : IDisposable where T : struct
```

Comment:
TestTable in current form is very deficient for testing more complex operations. As a requirements I would propose:

1. Support all input data formats for all HW intrinsic operations - Vector128, Vector256, numerical types, an even strings for SSE42.
2. Store output of intrinsic under test as well as output of verification function alongside test data.
3. Allow for priniting to output of all collected test data for detailed verification.

Since we may need to have `TestTable` with 4 or even 5 data arrays it could be necessary to create set of `structs` or just switch to `class` and extend it via inheritance.

Reply:
I have some local work here that I haven't pushed yet.

This was basically renamed to `SimpleTestTable` and is meant for tests which take expect the same operation to be performed for each input. There are tests in other categories that would require their own new `TestTables`.

For example, some tests only operate on a single input, some operate on strings or scalar values, and some (like `Shuffle` or `Unpack`) require checking `inArray1`, `inArray2`, and `outArray` as a whole, rather than an element at time.

Reply:
I have some local work as well - I hope I can push it soon. It has all this difficult operations tested with several input - ouput combinations.